Fix: Show loading state in WC input between pairing and receiving proposal

jmealy opened the pull request:
## What it solves

Resolves #3598
Resolves #3731

## How this PR fixes it
After successfully pairing with wallet connect, and before receiving a proposal, keep the paste button in its loading state.
Only remove the loading state after the proposal is received and handled.

## How to test it
note: This bug only appears when Walletconnect is slow to send the proposal from their side.
- connect a safe to any dapp using walletconnect.
- after pasting the pairing code, the loading state should appear.
- the loading state should persist until the WC modal closes.

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [ ] I've written a unit/e2e test for it (if applicable) 🧑‍💻

usame-algan reviewed (changes requested):
We are disabling certain buttons when loading is set to true so I think we need to set loading to undefined in `WcSessionManager` as soon as we receive a proposal otherwise the user can't click anything

<img width="524" alt="Screenshot 2024-05-16 at 10 19 51" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/72a34adc-bc88-409d-92fb-66fe1f95cf22">

usame-algan commented:
As a future idea: We could also add a timer for the loading state and give the user more feedback if it takes too long after connecting and not receiving a proposal or when trying to use an invalid wc uri. Right now we are showing an infinite loading state until the user reloads the app.

jmealy commented:
> Yes, the change Usame requested is necessary for this to pass. I'll kick it back.
> 
> Also I don't think the timeout should be something to do in the future. Maybe a 30 secs timeout at least?

I fixed the disabled buttons and added in the timeout of 30 seconds. After it times out it shows this error:

<img width="469" alt="image" src="https://github.com/safe-global/safe-wallet-web/assets/17801424/44277793-4d08-4f1a-8d1d-ed079f187a04">

usame-algan reviewed (approved):
Nice, thanks for implementing the timeout! 🚀 

@francovenica could you do a last check with the timeout? Good to go from my side!

francovenica commented:
LGTM
